<a href="https://colab.research.google.com/github/AyraCaps/Dsa-arya/blob/main/ASSIGNMENT_NEW.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## *Assignment on Measures of Central Tendency and Dispersion*

---



In [ ]:
import pandas as pd

In [ ]:
# Load the dataset
data=pd.read_csv('/content/StudentsPerformance.csv')
data.head()   # Display the first few rows of the dataset

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


***1.Males and females participated in the test.***

In [ ]:
# Number of males and females participated in the test
data['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64



*   Number of males participated in the test= 518
*   Number of females participated in the test= 482
  



***2.The students' parental level of education***

In [ ]:
# display the parental level of education of the students
data['parental level of education'].unique()

array(['some high school', 'high school', 'some college',
       "associate's degree", "bachelor's degree", "master's degree"],
      dtype=object)

In [ ]:
# average marks of students based on their parental level of eduction
parent_education=data.groupby('parental level of education').mean()
parent_education.sort_values(['math score','reading score','writing score'],ascending=False)

<ipython-input-12-3ef7065c38ca>:2: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  parent_education=data.groupby('parental level of education').mean()


math score  reading score  writing score
parental level of education                                          
master's degree               70.254237      75.949153      75.677966
bachelor's degree             69.288136      73.000000      73.381356
associate's degree            67.977477      71.018018      70.031532
some college                  67.128319      69.566372      69.035398
some high school              63.134078      66.759777      64.888268
high school                   61.821429      64.602041      62.403061



*   Parental level of education: from high school to a master’s degree


*   Students  with a master’s degree as the parental level of education have highest average scores in maths , readind and writing.

*   Students  with  high school as the parental level of education have lowest average scores in maths , readind and writing.






***3. Average score on each subject w.r.t gender and Test preparation course***

In [ ]:
#average score based on gender
data.groupby('gender')[['math score','reading score','writing score']].mean()

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058


*   Female students have higher average
reading and writing scores than their male counterparts. Where male students have higher average math scores

In [ ]:
#average score based on test preparation course
data.groupby('test preparation course').mean()

<ipython-input-8-e468d0dc9a30>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  data.groupby('test preparation course').mean()


math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

* Those who completed the Test preparation course scores higher average score for math, reading and writing than the ones who did not take any test preparation course.

***4. Scoring variation based on gender and Test preparation course***

Based on the standard deviation ,which is a measure of the variability or dispersion of scores within a dataset we can find the relative performance of students based on their gender and test preparation course.

In [ ]:
data.groupby('gender').std()

<ipython-input-11-bc5368fe0e7a>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.std is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  data.groupby('gender').std()


math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225



*    Standard deviation of scores of the male students is low, which indicates less variability. So they are more consistent in their performance than female students.



In [ ]:
data.groupby('test preparation course').std()

<ipython-input-10-9803286904b5>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.std is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  data.groupby('test preparation course').std()


math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667



*   Standard deviation of scores of the  students with those who completed the Test preparation course is low, which indicates less variability. So they are more consistent in their performance than the ones who did not take any test preparation course.





***5.The top 25% of students based on their Math score***





In [ ]:
#Select the students with Math score in the top 25%
maths_top=data['math score'].quantile(0.75)
Top25_Percent=data[data['math score']>=maths_top]
# students getting bonus points
bonus_point_students=Top25_Percent.sort_values('math score',ascending=False)
bonus_point_students

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
998    male        group E          associate's degree  free/reduced   
997    male        group D                some college      standard   
996    male        group A                some college      standard   
517  female        group E          associate's degree      standard   
..      ...            ...                         ...           ...   
854    male        group D            some high school      standard   
855    male        group E          associate's degree  free/reduced   
856    male        group E            some high school      standard   
415  female        group B             master's degree  free/reduced   
414  female        group A                 high school  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
998               completed         100            100             93  
997               completed         100             97             99  
996               completed         100             96             86  
517                    none         100            100            100  
..                      ...         ...            ...            ...  
854               completed          77             68             69  
855               completed          77             69             68  
856               completed          77             76             77  
415               completed          77             97             94  
414               completed          77             88             85  

[255 rows x 8 columns]